Modules/_json.c

```diff
@@ -1866,25 +1814,59 @@ PyDoc_STRVAR(module_doc,
 static int
 _json_exec(PyObject *module)
 {
-    if (PyType_Ready(&PyScannerType) < 0) {
+    PyObject *PyScannerType = PyType_FromSpec(&PyScannerType_spec);
+    if (PyScannerType == NULL) {
         return -1;
     }
-    if (PyType_Ready(&PyEncoderType) < 0) {
+    get_json_state(module)->PyScannerType = PyScannerType;
+    Py_INCREF(PyScannerType);
+    if (PyModule_AddObject(module, "make_scanner", get_json_state(module)->PyScannerType) < 0) {
         return -1;
     }
-    Py_INCREF((PyObject*)&PyScannerType);
-    if (PyModule_AddObject(module, "make_scanner", (PyObject*)&PyScannerType) < 0) {
-        Py_DECREF((PyObject*)&PyScannerType);
+
+    PyObject *PyEncoderType = PyType_FromSpec(&PyEncoderType_spec);
+    if (PyEncoderType == NULL) {
         return -1;
     }
-    Py_INCREF((PyObject*)&PyEncoderType);
-    if (PyModule_AddObject(module, "make_encoder", (PyObject*)&PyEncoderType) < 0) {
-        Py_DECREF((PyObject*)&PyEncoderType);
+    get_json_state(module)->PyEncoderType = PyEncoderType;
+    Py_INCREF(PyEncoderType);
+    if (PyModule_AddObject(module, "make_encoder", get_json_state(module)->PyEncoderType) < 0) {
         return -1;
     }
     return 0;
 }
 
+static int
+_jsonmodule_traverse(PyObject *module, visitproc visit, void *arg)
+{
+    _jsonmodulestate *state = (_jsonmodulestate *)PyModule_GetState(module);
+    if (state) {
```

Comment:
state cannot be NULL since https://bugs.python.org/issue39824: remove the if. Same remark for clear.